modules/nextflow/src/main/groovy/nextflow/container/CharliecloudCache.groovy

```diff
@@ -114,10 +114,10 @@ class CharliecloudCache {
     @PackageScope
     Path getCacheDir() {
 
-        if( config.pullTimeout )
-            pullTimeout = config.pullTimeout as Duration
+        if( config['pullTimeout'] )
+            pullTimeout = config['pullTimeout'] as Duration
```

Comment:
As this point is better to add public getters for each of these returning the expected type

Reply:
As groovy4 is an interesting feature but not urgent, I can mark this PR as a draft and try to maintain it updated (rebasing from master frequently) 

in this way, we can fix this kind of issue not directly related to groovy4 in other branches